### PR TITLE
display point cloud in dreamview

### PR DESCRIPTION
**Pull request by unacao**

### CODE REVIEW

File: modules/dreamview/backend/point_cloud/point_cloud_updater.cc [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+void PointCloudUpdater::UpdatePointCloud(const PointCloud2 &point_cloud) {
+  last_receive_time_ = Clock::NowInSeconds();
+  // transform from ros to pcl
+  pcl::PointCloud<pcl::PointXYZ> pcl_data;
+  pcl::fromROSMsg(point_cloud, pcl_data);
+
+  point_cloud_.Clear();
+  int count = 0;
+  for (size_t idx = 0; idx < pcl_data.size(); idx += 4) {
```
Comment (siyangy) [inline modules/dreamview/backend/point_cloud/point_cloud_updater.cc]: Let's make this a const member to be more prominent.

File: modules/dreamview/backend/point_cloud/point_cloud_updater.cc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    pcl::PointXYZ& pt = pcl_data.points[idx];
+    if (!isnan(pt.x) && !isnan(pt.y) && !isnan(pt.z)) {
+      count++;
+      point_cloud_.add_num(pt.x);
+      point_cloud_.add_num(pt.y);
+      point_cloud_.add_num(pt.z + 1.91);
```
Comment (siyangy) [inline modules/dreamview/backend/point_cloud/point_cloud_updater.cc]: Are we able to get the magic number from some config? Different car model would have different height.

File: modules/tools/supervisord/dev.conf [hunk 3]
```diff
@@ -116,7 +116,7 @@ redirect_stderr=true
 stdout_logfile=/apollo/data/log/dreamview.out
 
 [program:sim_control]
-command=/apollo/bazel-bin/modules/dreamview/dreamview --flagfile=/apollo/modules/dreamview/conf/dreamview.conf --enable_sim_control --log_dir=/apollo/data/log
+command=/apollo/bazel-bin/modules/dreamview/dreamview --flagfile=/apollo/modules/dreamview/conf/dreamview.conf --log_dir=/apollo/data/log
```
Comment (siyangy) [inline modules/tools/supervisord/dev.conf]: this section can be removed as a whole, as well as in release.conf.

File: modules/dreamview/backend/point_cloud/point_cloud_updater.cc [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        {
+          boost::shared_lock<boost::shared_mutex> reader_lock(mutex_);
+          to_send = point_cloud_str_;
+        }
+        websocket_->SendBinaryData(conn, to_send, true);;
+      });
+}
+
+void PointCloudUpdater::Start() {
+  AdapterManager::AddPointCloudCallback(&PointCloudUpdater::UpdatePointCloud,
```
Comment (siyangy) [inline modules/dreamview/backend/point_cloud/point_cloud_updater.cc]: Do we need to check whether the PointCloudAdapter is present?

File: modules/dreamview/backend/point_cloud/point_cloud_updater.cc (hunk 2 above)
Comment (Capri2014) [reply modules/dreamview/backend/point_cloud/point_cloud_updater.cc]: @unacao @siyangy Is that case put it in https://github.com/ApolloAuto/apollo/blob/master/modules/common/configs/proto/vehicle_config.proto maybe? This is where we usually load vehicle related configs.

Comment (siyangy) [reply modules/dreamview/backend/point_cloud/point_cloud_updater.cc]: looks like it's from velodyne64_height.yaml, not sure we have a programatic way to read it though.

**Pull request merged**